main/src/library/Map.flix

```diff
@@ -126,19 +144,61 @@ namespace Map {
     ///
     /// Returns `None` if `m` is empty.
     ///
-    pub def minimum(m: Map[k, v]): Option[(k, v)] =
-        let Map(xs) = m;
-        RedBlackTree.minimum(xs)
+    pub def minimumKey(m: Map[k, v]): Option[(k, v)] =
+        let Map(t) = m;
+        RedBlackTree.minimum(t)
+
+    ///
+    /// Optionally finds `k -> v` where `k` is the smallest key according to the given comparator `cmp`.
+    ///
+    /// Returns `None` if `m` is empty.
+    ///
+    /// Purity polymorphic: Runs in parallel when given a pure function `cmp`.
+    ///
+    @ParallelWhenPure
+    pub def minimumKeyBy(cmp: (k, k) -> Comparison & ef, m: Map[k, v]): Option[(k, v)] & ef =
+        let Map(t) = m;
+        let f = r -> cmp(r.kl, r.kr);
+        reifyEff(f) {
+            case Pure(g) =>
+                let h = (kl, _, kr, _) -> g({kl = kl, kr = kr});
+                if (useParallelEvaluation(m))
+                    RedBlackTree.parMinimumBy(threads() - 1, h, t) as & Pure
+                else
+                    RedBlackTree.reduceLeft((kl, vl, kr, vr) -> if (g({kl = kl, kr = kr}) == LessThan) (kl, vl) else (kr, vr), t)
+            case _ => RedBlackTree.reduceLeft((kl, vl, kr, vr) -> if (cmp(kl, kr) == LessThan) (kl, vl) else (kr, vr), t)
```

Comment:
- [ ] Use `Map.reduceLeft` directly.
- [ ] You don't have to use `g` in the else case. You can just use `cmp` directly with `reduceLeft`.

I think this would simplify the code.

Reply:
This comment also applies in a few other places.